subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/versionupgrade/UpgradeDependencyPlugin.java

```diff
@@ -254,6 +256,37 @@ public void execute(String openPullRequestBranch) {
             }
         });
 
+        TaskMaker.task(project, MERGE_PULL_REQUEST, MergePullRequestTask.class, new Action<MergePullRequestTask>() {
+            @Override
+            public void execute(final MergePullRequestTask task) {
+                task.setDescription("Merge pull request when all checks will be passed");
+                task.mustRunAfter(CREATE_PULL_REQUEST);
+                task.setGitHubApiUrl(conf.getGitHub().getApiUrl());
+                task.setDryRun(conf.isDryRun());
+                task.setAuthToken(conf.getLenient().getGitHub().getWriteAuthToken());
+                task.setVersionUpgrade(upgradeDependencyExtension);
+
+                gitOriginPlugin.provideOriginRepo(task, new Action<String>() {
+                    @Override
+                    public void execute(String originRepoName) {
+                        task.setForkRepositoryName(originRepoName);
+                        task.setUpstreamRepositoryName(conf.getGitHub().getRepository());
+                    }
+                });
+
+                findOpenPullRequestTask.provideOpenPullRequest(task, new Action<OpenPullRequest>() {
```

Comment:
What about the case when there is no open pull request? I think we have to have two different modes (doesn't have to be implemented now) - one when we update the PR and one when we automatically merge. It doesn't make sense to use both.

So we have to add providePullRequest method to createPullRequest task too. This way the pull request to merge will be provided either by findOpenPullRequest task or createPullRequest (when there is no open one).

Reply:
@wwilk I pushed my proposition of solution. Please take a look